File: cucumber-expressions/java/src/test/java/io/cucumber/cucumberexpressions/ArgumentTest.java
package io.cucumber.cucumberexpressions;

import org.junit.jupiter.api.Test;

import java.util.Collections;
import java.util.List;
import java.util.Locale;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class ArgumentTest {
    @Test
    public void exposes_parameter_type() {
        TreeRegexp treeRegexp = new TreeRegexp("three (.*) mice");
        ParameterTypeRegistry parameterTypeRegistry = new ParameterTypeRegistry(Locale.ENGLISH);
        List<Argument<?>> arguments = Argument.build(treeRegexp, "three blind mice", Collections.singletonList(parameterTypeRegistry.lookupByTypeName("string")));
        Argument<?> argument = arguments.get(0);
        assertEquals("string", argument.getParameterType().getName());
    }
}

File: cucumber-expressions/javascript/test/ArgumentTest.ts
import TreeRegexp from '../src/TreeRegexp'
import ParameterTypeRegistry from '../src/ParameterTypeRegistry'
import Argument from '../src/Argument'
import * as assert from 'assert'

describe('Argument', () => {
  it('exposes getParameterTypeName()', () => {
    const treeRegexp = new TreeRegexp('three (.*) mice')
    const parameterTypeRegistry = new ParameterTypeRegistry()
    const args = Argument.build(treeRegexp, 'three blind mice', [
      parameterTypeRegistry.lookupByTypeName('string'),
    ])
    const argument = args[0]
    assert.strictEqual(argument.getParameterType().name, 'string')
  })
})
